### DISCUSSION TITLE
Setting up fluvio in windows environment

### DISCUSSION THREAD

**Discussion by kharabela** [fluvio-community/fluvio#3880] [Q&A] [answered]
Is there a guide to setup fluvio in Windows environment?

**Comment by drc-infinyon** [comment, marked as answer]
You need to use WSL that is Windows Subsystem for Linux. Follow this guide: https://www.fluvio.io/docs/get-started/linux/

**Reply by kharabela** [reply]
curl -fsS https://hub.infinyon.cloud/install/install.sh | bash
curl: (60) SSL certificate problem: unable to get local issuer certificate
More details here: https://curl.se/docs/sslcerts.html

curl failed to verify the legitimacy of the server and therefore could not
establish a secure connection to it. To learn more about this situation and
how to fix it, please visit the web page mentioned above.

**Reply by drc-infinyon** [reply]
Not an issue with our script. It's most likely configuration issue. It is failing as cURL is unable to verify the certificate provided by the server.

Need to reproduce the issue on a windows system. We are all on macs, linux. Figuring out a workaround. Have created an issue to track. https://github.com/infinyon/fluvio/issues/3881.

**Reply by kharabela** [reply]
Wanna ask one thing.Rust binaries work fine in windows,why force users to use curl instead of windows binaries to run

**Reply by drc-infinyon** [reply]
A single script installer has worked well so far. Our install was a bit more complex a few months ago and we have been simplifying.

We have not had many Windows users so far (~3 in roughly a year). 

Event though rust binaries work on Windows. the tooling has its quirks to support. Deployments are predominantly Linux builds in Kubernetes containers.

We have have not had enough Windows users to need to work this differently yet.

**Reply by drc-infinyon** [reply]
Hi @kharabela can you try this script to fix the certificates:

https://github.com/bayaro/windows-certs-2-wsl

**Reply by kharabela** [reply]
@drc-infinyon -retried on windows 11 docker
PS C:\Fluvio> cd examples
PS C:\Fluvio\examples> cd docker-compose
PS C:\Fluvio\examples\docker-compose> docker compose up
[+] Running 4/4
 ✔ sc Pulled                                                                                                                                                     6.6s 
 ✔ spu 2 layers [⣿⣿]      0B/0B      Pulled                                                                                                                      6.6s 
   ✔ f7dab3ab2d6e Already exists                                                                                                                                 0.0s 
   ✔ c9626b09e02d Pull complete                                                                                                                                  2.4s 
[+] Building 29.4s (8/8) FINISHED                                                                                                                      docker:default
 => [sc-setup internal] load build definition from Dockerfile                                                                                                    0.2s
 => => transferring dockerfile: 253B                                                                                                                             0.0s 
 => [sc-setup internal] load metadata for docker.io/library/ubuntu:20.04                                                                                         1.8s
 => [sc-setup internal] load .dockerignore                                                                                                                       0.1s
 => => transferring context: 2B                                                                                                                                  0.0s 
 => [sc-setup 1/4] FROM docker.io/library/ubuntu:20.04@sha256:80ef4a44043dec4490506e6cc4289eeda2d106a70148b74b5ae91ee670e9c35d                                   4.2s 
 => => resolve docker.io/library/ubuntu:20.04@sha256:80ef4a44043dec4490506e6cc4289eeda2d106a70148b74b5ae91ee670e9c35d                                            0.0s 
 => => sha256:80ef4a44043dec4490506e6cc4289eeda2d106a70148b74b5ae91ee670e9c35d 1.13kB / 1.13kB                                                                   0.0s
 => => sha256:48c35f3de33487442af224ed4aabac19fd9bfbd91ee90e9471d412706b20ba73 424B / 424B                                                                       0.0s 
 => => sha256:3cff1c6ff37e0101a08119d0743ba95c7f505d00298f5a169129e4e9086cde9e 2.29kB / 2.29kB                                                                   0.0s 
 => => sha256:17d0386c2fff30a5b92652bbef2b84639dba9b9f17bdbb819c8d10badd827fdb 27.51MB / 27.51MB                                                                 2.7s 
 => => extracting sha256:17d0386c2fff30a5b92652bbef2b84639dba9b9f17bdbb819c8d10badd827fdb                                                                        1.2s
 => [sc-setup 2/4] RUN apt-get update                                                                                                                            7.9s
 => [sc-setup 3/4] RUN apt-get install -y curl unzip                                                                                                            14.0s
 => [sc-setup 4/4] RUN curl -fsS https://hub.infinyon.cloud/install/install.sh?ctx=dc | bash                                                                     0.8s
 => [sc-setup] exporting to image                                                                                                                                0.2s 
 => => exporting layers                                                                                                                                          0.2s 
 => => writing image sha256:e26214c99e9be21538d907e6dfd7bb41c85eefe9c0e64176c8bd8751054005ee                                                                     0.0s 
 => => naming to docker.io/library/docker-compose-sc-setup                                                                                                       0.0s 
[+] Running 4/4
 ✔ Network docker-compose_default  Created                                                                                                                       0.0s 
 ✔ Container sc                    Created                                                                                                                       0.1s 
 ✔ Container spu                   Created                                                                                                                       0.1s 
 ✔ Container sc-setup              Created                                                                                                                       0.1s 
Attaching to sc, sc-setup, spu
sc        | CLI Option: ScOpt {
sc        |     run_mode: ScOptRunMode {
sc        |         local: Some(
sc        |             "/fluvio/metadata",
sc        |         ),
sc        |         k8: false,
sc        |         read_only: None,
sc        |     },
sc        |     bind_public: None,
sc        |     bind_private: None,
sc        |     namespace: None,
sc        |     tls: TlsConfig {
sc        |         tls: false,
sc        |         server_cert: None,
sc        |         server_key: None,
sc        |         enable_client_cert: false,
sc        |         ca_cert: None,
sc        |         bind_non_tls_public: None,
sc        |         secret_name: None,
sc        |     },
sc        |     x509_auth_scopes: None,
sc        |     auth_policy: None,
sc        |     white_list: [],
sc        | }
sc        | 2024-03-27T16:16:02.219373Z  INFO fluvio_sc::start: Platform version="0.11.5"
sc        | 2024-03-27T16:16:02.219417Z  INFO fluvio_sc::start: Git commit="b500708703b0c8900675af56f4fa03e3ae49664c"
sc        | 2024-03-27T16:16:02.219454Z  INFO fluvio_sc::start: System name=Some("Alpine Linux")
sc        | 2024-03-27T16:16:02.219466Z  INFO fluvio_sc::start: System kernel=Some("5.15.133.1-microsoft-standard-WSL2")
sc        | 2024-03-27T16:16:02.219477Z  INFO fluvio_sc::start: System os_version=Some("Linux 3.14.10 Alpine Linux")
sc        | 2024-03-27T16:16:02.219760Z  INFO fluvio_sc::start: System core_count=Some(10)
sc        | 2024-03-27T16:16:02.219793Z  INFO fluvio_sc::start: System total_memory=16597237760
sc        | 2024-03-27T16:16:02.219799Z  INFO fluvio_sc::start: System available_memory=16593932288
sc        | 2024-03-27T16:16:02.219820Z  INFO fluvio_sc::start: Uptime in secs uptime=656
sc        | Starting SC, platform: 0.11.5
sc        | 2024-03-27T16:16:02.219882Z  INFO fluvio_sc::start: Running in local mode metadata="/fluvio/metadata"
sc        | 2024-03-27T16:16:02.222344Z  INFO fluvio_sc::start: starting local main loop
sc        | 2024-03-27T16:16:02.222631Z  INFO fluvio_sc::controllers::spus::controller: starting spu controller
sc        | 2024-03-27T16:16:02.222798Z  INFO sc_private_server{address="0.0.0.0:9004"}: fluvio_sc::services::private_api: starting internal services
sc        | 2024-03-27T16:16:02.222817Z  INFO SpuControllerLoop: fluvio_sc::controllers::spus::controller: started
sc        | 2024-03-27T16:16:02.222825Z  INFO fluvio_sc::init::pub_server: using root authorization
sc        | Streaming Controller started successfully
sc        | 2024-03-27T16:16:02.223127Z  INFO PartitionController: fluvio_sc::controllers::partitions::controller: started
sc        | 2024-03-27T16:16:02.223576Z  INFO accept_incoming{self=FluvioApiServer("0.0.0.0:9004")}: fluvio_service::server: Opened TcpListener, waiting for connections
sc        | 2024-03-27T16:16:02.223649Z  INFO accept_incoming{self=FluvioApiServer("0.0.0.0:9003")}: fluvio_service::server: Opened TcpListener, waiting for connections
sc-setup  | /bin/sh: 1: fluvio: not found
sc-setup  | /bin/sh: 1: fluvio: not found
spu       | 2024-03-27T16:16:02.941349Z  INFO fluvio_spu::config::cli: using sc endpoint from env var: sc:9004
spu       | 2024-03-27T16:16:02.941398Z  INFO fluvio_spu::config::cli: overriding log base: /fluvio/data
spu       | 2024-03-27T16:16:02.941403Z  INFO fluvio_spu::config::cli: overriding public addr: spu:9010
spu       | 2024-03-27T16:16:02.941406Z  INFO fluvio_spu::config::cli: overriding private addr: spu:9011
spu       | starting spu server (id:5001)
spu       | 2024-03-27T16:16:02.999026Z  INFO fluvio_spu::start: Platform version="0.11.5"
spu       | 2024-03-27T16:16:02.999062Z  INFO fluvio_spu::start: Git commit="b500708703b0c8900675af56f4fa03e3ae49664c"
spu       | 2024-03-27T16:16:02.999085Z  INFO fluvio_spu::start: System name=Some("Alpine Linux")
spu       | 2024-03-27T16:16:02.999093Z  INFO fluvio_spu::start: System kernel=Some("5.15.133.1-microsoft-standard-WSL2")
spu       | 2024-03-27T16:16:02.999101Z  INFO fluvio_spu::start: System os_version=Some("Linux 3.14.10 Alpine Linux")
spu       | 2024-03-27T16:16:02.999293Z  INFO fluvio_spu::start: System core_count=Some(10)
spu       | 2024-03-27T16:16:02.999311Z  INFO fluvio_spu::start: System total_memory=16597237760
spu       | 2024-03-27T16:16:02.999315Z  INFO fluvio_spu::start: System available_memory=16594305024
spu       | 2024-03-27T16:16:02.999325Z  INFO fluvio_spu::start: Uptime in secs uptime=656
spu       | 2024-03-27T16:16:03.002767Z  INFO fluvio_spu::services::public: Starting SPU public service: spu_id=5001 addr=spu:9010
spu       | 2024-03-27T16:16:03.002826Z  INFO fluvio_spu::services::internal: starting SPU: 5001 at internal service at: spu:9011
spu       | SPU Version: 0.0.0 started successfully
spu       | 2024-03-27T16:16:03.003983Z  INFO fluvio_spu::control_plane::dispatcher: trying to create socket to sc sc_endpoint=sc:9004 spu_id=5001
spu       | 2024-03-27T16:16:03.004065Z  INFO fluvio_spu::monitoring: using default metric path: /tmp/fluvio-spu.sock
spu       | 2024-03-27T16:16:03.004592Z  INFO accept_incoming{self=FluvioApiServer("spu:9010")}: fluvio_service::server: Opened TcpListener, waiting for connections  
spu       | 2024-03-27T16:16:03.004896Z  INFO accept_incoming{self=FluvioApiServer("spu:9011")}: fluvio_service::server: Opened TcpListener, waiting for connections  
spu       | 2024-03-27T16:16:03.005349Z  INFO fluvio_spu::monitoring: monitoring started
sc        | 2024-03-27T16:16:03.006954Z  INFO accept_incoming{self=FluvioApiServer("0.0.0.0:9004")}: fluvio_service::server: Received connection, spawning request handler
spu       | 2024-03-27T16:16:03.007406Z  INFO fluvio_spu::control_plane::dispatcher: connected to sc for spu spu_id=5001
spu       | 2024-03-27T16:16:03.007453Z  INFO fluvio_spu::control_plane::dispatcher: established connection to sc for spu local_spu_id=5001
sc        | 2024-03-27T16:16:03.008059Z  INFO handle_request{host="0.0.0.0:9004"}: fluvio_service::server: Response sent successfully, closing connection host=0.0.0.0:9004 peer_addr=172.18.0.4:38982
spu       | 2024-03-27T16:16:03.008157Z  WARN send_spu_registration{socket=10}: fluvio_spu::control_plane::dispatcher: spu registration failed err="registeration failed" local_spu_id=5001
spu       | 2024-03-27T16:16:03.008256Z  WARN fluvio_spu::control_plane::dispatcher: sleeping 3 seconds before re-trying re-register
sc-setup exited with code 0
spu       | 2024-03-27T16:16:06.008456Z  INFO fluvio_spu::control_plane::dispatcher: trying to create socket to sc sc_endpoint=sc:9004 spu_id=5001
spu       | 2024-03-27T16:16:06.009194Z  INFO fluvio_spu::control_plane::dispatcher: connected to sc for spu spu_id=5001
sc        | 2024-03-27T16:16:06.009196Z  INFO accept_incoming{self=FluvioApiServer("0.0.0.0:9004")}: fluvio_service::server: Received connection, spawning request handler
spu       | 2024-03-27T16:16:06.009216Z  INFO fluvio_spu::control_plane::dispatcher: established connection to sc for spu local_spu_id=5001
sc        | 2024-03-27T16:16:06.009346Z  INFO handle_request{host="0.0.0.0:9004"}: fluvio_service::server: Response sent successfully, closing connection host=0.0.0.0:9004 peer_addr=172.18.0.4:52150
spu       | 2024-03-27T16:16:06.009375Z  WARN send_spu_registration{socket=10}: fluvio_spu::control_plane::dispatcher: spu registration failed err="registeration failed" local_spu_id=5001
spu       | 2024-03-27T16:16:06.009405Z  WARN fluvio_spu::control_plane::dispatcher: sleeping 3 seconds before re-trying re-register
spu       | 2024-03-27T16:16:09.009596Z  INFO fluvio_spu::control_plane::dispatcher: trying to create socket to sc sc_endpoint=sc:9004 spu_id=5001
sc        | 2024-03-27T16:16:09.010381Z  INFO accept_incoming{self=FluvioApiServer("0.0.0.0:9004")}: fluvio_service::server: Received connection, spawning request handler
spu       | 2024-03-27T16:16:09.010435Z  INFO fluvio_spu::control_plane::dispatcher: connected to sc for spu spu_id=5001
spu       | 2024-03-27T16:16:09.010457Z  INFO fluvio_spu::control_plane::dispatcher: established connection to sc for spu local_spu_id=5001
sc        | 2024-03-27T16:16:09.010659Z  INFO handle_request{host="0.0.0.0:9004"}: fluvio_service::server: Response sent successfully, closing connection host=0.0.0.0:9004 peer_addr=172.18.0.4:52154
spu       | 2024-03-27T16:16:09.010712Z  WARN send_spu_registration{socket=10}: fluvio_spu::control_plane::dispatcher: spu registration failed err="registeration failed" local_spu_id=5001
spu       | 2024-03-27T16:16:09.010742Z  WARN fluvio_spu::control_plane::dispatcher: sleeping 3 seconds before re-trying re-register
spu       | 2024-03-27T16:16:12.011133Z  INFO fluvio_spu::control_plane::dispatcher: trying to create socket to sc sc_endpoint=sc:9004 spu_id=5001
spu       | 2024-03-27T16:16:12.012504Z  INFO fluvio_spu::control_plane::dispatcher: connected to sc for spu spu_id=5001
spu       | 2024-03-27T16:16:12.012526Z  INFO fluvio_spu::control_plane::dispatcher: established connection to sc for spu local_spu_id=5001
sc        | 2024-03-27T16:16:12.012260Z  INFO accept_incoming{self=FluvioApiServer("0.0.0.0:9004")}: fluvio_service::server: Received connection, spawning request handler
spu       | 2024-03-27T16:16:12.012802Z  WARN send_spu_registration{socket=10}: fluvio_spu::control_plane::dispatcher: spu registration failed err="registeration failed" local_spu_id=5001
sc        | 2024-03-27T16:16:12.012729Z  INFO handle_request{host="0.0.0.0:9004"}: fluvio_service::server: Response sent successfully, closing connection host=0.0.0.0:9004 peer_addr=172.18.0.4:52160
spu       | 2024-03-27T16:16:12.012827Z  WARN fluvio_spu::control_plane::dispatcher: sleeping 3 seconds before re-trying re-register
spu       | 2024-03-27T16:16:15.013277Z  INFO fluvio_spu::control_plane::dispatcher: trying to create socket to sc sc_endpoint=sc:9004 spu_id=5001
sc        | 2024-03-27T16:16:15.014445Z  INFO accept_incoming{self=FluvioApiServer("0.0.0.0:9004")}: fluvio_service::server: Received connection, spawning request handler
spu       | 2024-03-27T16:16:15.014640Z  INFO fluvio_spu::control_plane::dispatcher: connected to sc for spu spu_id=5001
sc        | 2024-03-27T16:16:15.014854Z  INFO handle_request{host="0.0.0.0:9004"}: fluvio_service::server: Response sent successfully, closing connection host=0.0.0.0:9004 peer_addr=172.18.0.4:54634
spu       | 2024-03-27T16:16:15.014661Z  INFO fluvio_spu::control_plane::dispatcher: established connection to sc for spu local_spu_id=5001
spu       | 2024-03-27T16:16:15.015048Z  WARN send_spu_registration{socket=10}: fluvio_spu::control_plane::dispatcher: spu registration failed err="registeration failed" local_spu_id=5001
spu       | 2024-03-27T16:16:15.015071Z  WARN fluvio_spu::control_plane::dispatcher: sleeping 3 seconds before re-trying re-register
spu       | 2024-03-27T16:16:18.015567Z  INFO fluvio_spu::control_plane::dispatcher: trying to create socket to sc sc_endpoint=sc:9004 spu_id=5001
spu       | 2024-03-27T16:16:18.016923Z  INFO fluvio_spu::control_plane::dispatcher: connected to sc for spu spu_id=5001
spu       | 2024-03-27T16:16:18.016958Z  INFO fluvio_spu::control_plane::dispatcher: established connection to sc for spu local_spu_id=5001
sc        | 2024-03-27T16:16:18.017007Z  INFO accept_incoming{self=FluvioApiServer("0.0.0.0:9004")}: fluvio_service::server: Received connection, spawning request handler
sc        | 2024-03-27T16:16:18.017143Z  INFO handle_request{host="0.0.0.0:9004"}: fluvio_service::server: Response sent successfully, closing connection host=0.0.0.0:9004 peer_addr=172.18.0.4:54640
spu       | 2024-03-27T16:16:18.017391Z  WARN send_spu_registration{socket=10}: fluvio_spu::control_plane::dispatcher: spu registration failed err="registeration failed" local_spu_id=5001
spu       | 2024-03-27T16:16:18.017442Z  WARN fluvio_spu::control_plane::dispatcher: sleeping 3 seconds before re-trying re-register
spu       | 2024-03-27T16:16:21.018115Z  INFO fluvio_spu::control_plane::dispatcher: trying to create socket to sc sc_endpoint=sc:9004 spu_id=5001
sc        | 2024-03-27T16:16:21.019982Z  INFO accept_incoming{self=FluvioApiServer("0.0.0.0:9004")}: fluvio_service::server: Received connection, spawning request handler
spu       | 2024-03-27T16:16:21.020285Z  INFO fluvio_spu::control_plane::dispatcher: connected to sc for spu spu_id=5001
sc        | 2024-03-27T16:16:21.020889Z  INFO handle_request{host="0.0.0.0:9004"}: fluvio_service::server: Response sent successfully, closing connection host=0.0.0.0:9004 peer_addr=172.18.0.4:54656
spu       | 2024-03-27T16:16:21.020334Z  INFO fluvio_spu::control_plane::dispatcher: established connection to sc for spu local_spu_id=5001
spu       | 2024-03-27T16:16:21.020954Z  WARN send_spu_registration{socket=10}: fluvio_spu::control_plane::dispatcher: spu registration failed err="registeration failed" local_spu_id=5001
spu       | 2024-03-27T16:16:21.021020Z  WARN fluvio_spu::control_plane::dispatcher: sleeping 3 seconds before re-trying re-register
spu       | 2024-03-27T16:16:24.021899Z  INFO fluvio_spu::control_plane::dispatcher: trying to create socket to sc sc_endpoint=sc:9004 spu_id=5001
spu       | 2024-03-27T16:16:24.024384Z  INFO fluvio_spu::control_plane::dispatcher: connected to sc for spu spu_id=5001
spu       | 2024-03-27T16:16:24.024460Z  INFO fluvio_spu::control_plane::dispatcher: established connection to sc for spu local_spu_id=5001
sc        | 2024-03-27T16:16:24.024161Z  INFO accept_incoming{self=FluvioApiServer("0.0.0.0:9004")}: fluvio_service::server: Received connection, spawning request handler
spu       | 2024-03-27T16:16:24.025533Z  WARN send_spu_registration{socket=10}: fluvio_spu::control_plane::dispatcher: spu registration failed err="registeration failed" local_spu_id=5001
sc        | 2024-03-27T16:16:24.025029Z  INFO handle_request{host="0.0.0.0:9004"}: fluvio_service::server: Response sent successfully, closing connection host=0.0.0.0:9004 peer_addr=172.18.0.4:36564
spu       | 2024-03-27T16:16:24.025625Z  WARN fluvio_spu::control_plane::dispatcher: sleeping 3 seconds before re-trying re-register
spu       | 2024-03-27T16:16:27.026241Z  INFO fluvio_spu::control_plane::dispatcher: trying to create socket to sc sc_endpoint=sc:9004 spu_id=5001
spu       | 2024-03-27T16:16:27.027932Z  INFO fluvio_spu::control_plane::dispatcher: connected to sc for spu spu_id=5001
spu       | 2024-03-27T16:16:27.027975Z  INFO fluvio_spu::control_plane::dispatcher: established connection to sc for spu local_spu_id=5001
sc        | 2024-03-27T16:16:27.027629Z  INFO accept_incoming{self=FluvioApiServer("0.0.0.0:9004")}: fluvio_service::server: Received connection, spawning request handler
spu       | 2024-03-27T16:16:27.028418Z  WARN send_spu_registration{socket=10}: fluvio_spu::control_plane::dispatcher: spu registration failed err="registeration failed" local_spu_id=5001
spu       | 2024-03-27T16:16:27.028443Z  WARN fluvio_spu::control_plane::dispatcher: sleeping 3 seconds before re-trying re-register
sc        | 2024-03-27T16:16:27.028484Z  INFO handle_request{host="0.0.0.0:9004"}: fluvio_service::server: Response sent successfully, closing connection host=0.0.0.0:9004 peer_addr=172.18.0.4:36578
spu       | 2024-03-27T16:16:30.028920Z  INFO fluvio_spu::control_plane::dispatcher: trying to create socket to sc sc_endpoint=sc:9004 spu_id=5001
spu       | 2024-03-27T16:16:30.030807Z  INFO fluvio_spu::control_plane::dispatcher: connected to sc for spu spu_id=5001
spu       | 2024-03-27T16:16:30.030844Z  INFO fluvio_spu::control_plane::dispatcher: established connection to sc for spu local_spu_id=5001
sc        | 2024-03-27T16:16:30.030502Z  INFO accept_incoming{self=FluvioApiServer("0.0.0.0:9004")}: fluvio_service::server: Received connection, spawning request handler
spu       | 2024-03-27T16:16:30.031348Z  WARN send_spu_registration{socket=10}: fluvio_spu::control_plane::dispatcher: spu registration failed err="registeration failed" local_spu_id=5001
sc        | 2024-03-27T16:16:30.031211Z  INFO handle_request{host="0.0.0.0:9004"}: fluvio_service::server: Response sent successfully, closing connection host=0.0.0.0:9004 peer_addr=172.18.0.4:36594
spu       | 2024-03-27T16:16:30.031399Z  WARN fluvio_spu::control_plane::dispatcher: sleeping 3 seconds before re-trying re-register
spu       | 2024-03-27T16:16:33.031975Z  INFO fluvio_spu::control_plane::dispatcher: trying to create socket to sc sc_endpoint=sc:9004 spu_id=5001
spu       | 2024-03-27T16:16:33.033432Z  INFO fluvio_spu::control_plane::dispatcher: connected to sc for spu spu_id=5001
spu       | 2024-03-27T16:16:33.033456Z  INFO fluvio_spu::control_plane::dispatcher: established connection to sc for spu local_spu_id=5001
sc        | 2024-03-27T16:16:33.033241Z  INFO accept_incoming{self=FluvioApiServer("0.0.0.0:9004")}: fluvio_service::server: Received connection, spawning request handler
sc        | 2024-03-27T16:16:33.033837Z  INFO handle_request{host="0.0.0.0:9004"}: fluvio_service::server: Response sent successfully, closing connection host=0.0.0.0:9004 peer_addr=172.18.0.4:36610
spu       | 2024-03-27T16:16:33.033861Z  WARN send_spu_registration{socket=10}: fluvio_spu::control_plane::dispatcher: spu registration failed err="registeration failed" local_spu_id=5001
spu       | 2024-03-27T16:16:33.033891Z  WARN fluvio_spu::control_plane::dispatcher: sleeping 3 seconds before re-trying re-register
spu       | 2024-03-27T16:16:36.034542Z  INFO fluvio_spu::control_plane::dispatcher: trying to create socket to sc sc_endpoint=sc:9004 spu_id=5001
sc        | 2024-03-27T16:16:36.035900Z  INFO accept_incoming{self=FluvioApiServer("0.0.0.0:9004")}: fluvio_service::server: Received connection, spawning request handler
sc        | 2024-03-27T16:16:36.036436Z  INFO handle_request{host="0.0.0.0:9004"}: fluvio_service::server: Response sent successfully, closing connection host=0.0.0.0:9004 peer_addr=172.18.0.4:44414
spu       | 2024-03-27T16:16:36.036167Z  INFO fluvio_spu::control_plane::dispatcher: connected to sc for spu spu_id=5001
spu       | 2024-03-27T16:16:36.036189Z  INFO fluvio_spu::control_plane::dispatcher: established connection to sc for spu local_spu_id=5001
spu       | 2024-03-27T16:16:36.036500Z  WARN send_spu_registration{socket=10}: fluvio_spu::control_plane::dispatcher: spu registration failed err="registeration failed" local_spu_id=5001
spu       | 2024-03-27T16:16:36.036528Z  WARN fluvio_spu::control_plane::dispatcher: sleeping 3 seconds before re-trying re-register
spu       | 2024-03-27T16:16:39.037141Z  INFO fluvio_spu::control_plane::dispatcher: trying to create socket to sc sc_endpoint=sc:9004 spu_id=5001
sc        | 2024-03-27T16:16:39.038818Z  INFO accept_incoming{self=FluvioApiServer("0.0.0.0:9004")}: fluvio_service::server: Received connection, spawning request handler
spu       | 2024-03-27T16:16:39.039458Z  INFO fluvio_spu::control_plane::dispatcher: connected to sc for spu spu_id=5001
spu       | 2024-03-27T16:16:39.039490Z  INFO fluvio_spu::control_plane::dispatcher: established connection to sc for spu local_spu_id=5001
sc        | 2024-03-27T16:16:39.039821Z  INFO handle_request{host="0.0.0.0:9004"}: fluvio_service::server: Response sent successfully, closing connection host=0.0.0.0:9004 peer_addr=172.18.0.4:44430
spu       | 2024-03-27T16:16:39.039991Z  WARN send_spu_registration{socket=10}: fluvio_spu::control_plane::dispatcher: spu registration failed err="registeration failed" local_spu_id=5001
spu       | 2024-03-27T16:16:39.040032Z  WARN fluvio_spu::control_plane::dispatcher: sleeping 3 seconds before re-trying re-register
